ui/src/app/workflows/components/submit-workflow-panel.tsx

```diff
@@ -17,119 +17,92 @@ interface Props {
-        return (
-            <>
-                <h4>Submit Workflow</h4>
-                <h5>
-                    {this.props.namespace}/{this.props.name}
-                </h5>
-                {this.state.error && <ErrorNotice error={this.state.error} />}
-                <div className='white-box'>
-                    <div key='entrypoint' title='Entrypoint' style={{marginBottom: 25}}>
-                        <label>Entrypoint</label>
-                        <Select
-                            value={this.state.entrypoint}
-                            options={this.state.templates.map(t => ({
-                                value: t.name,
-                                title: t.name
-                            }))}
-                            onChange={selected => {
-                                const selectedTemplate = this.getSelectedTemplate(selected.value);
-                                this.setState({
-                                    entrypoint: selected.value,
-                                    selectedTemplate,
-                                    parameters: (selectedTemplate && selectedTemplate.inputs.parameters) || []
-                                });
-                            }}
-                        />
-                    </div>
-                    <div key='parameters' style={{marginBottom: 25}}>
-                        <label>Parameters</label>
-                        {this.state.workflowParameters.length > 0 && (
-                            <ParametersInput parameters={this.state.workflowParameters} onChange={workflowParameters => this.setState({workflowParameters})} />
-                        )}
-                        {this.state.parameters.length > 0 && <ParametersInput parameters={this.state.parameters} onChange={parameters => this.setState({parameters})} />}
-                        {this.state.workflowParameters.length === 0 && this.state.parameters.length === 0 ? (
-                            <>
-                                <br />
-                                <label>No parameters</label>
-                            </>
-                        ) : (
-                            <></>
-                        )}
-                    </div>
-                    <div key='labels' style={{marginBottom: 25}}>
-                        <label>Labels</label>
-                        <TagsInput tags={this.state.labels} onChange={labels => this.setState({labels})} />
-                    </div>
-                    <div key='submit'>
-                        <button onClick={() => this.submit()} className='argo-button argo-button--base' disabled={this.state.isSubmitting}>
-                            <i className='fa fa-plus' /> {this.state.isSubmitting ? 'Loading...' : 'Submit'}
-                        </button>
-                    </div>
-                </div>
-            </>
-        );
-    }
+    const [entrypoint, setEntrypoint] = useState<string>(workflowEntrypoint);
```

Comment:
```suggestion
    const [entrypoint, setEntrypoint] = useState(workflowEntrypoint);
```

`string` straightforwardly inferred here, so type annotation is not needed.

you didn't use type annotations for primitives in `resubmit-workflow-panel.tsx` and I didn't use them in my PRs, so let's keep it consistent and only use them when it's not correctly inferred (which tends to happen with more complex types or different initial states -- `Parameter[]` does need the annotation, for instance).